buildcheck/views/notification_page.py

```diff
@@ -0,0 +1,81 @@
+import reflex as rx
+from datetime import datetime
+from typing import List, Dict
+from buildcheck.backend.supabase_client import supabase_client
+# Importing reusable layout components created by the team 
+from buildcheck.components.navbar import navbar
+from buildcheck.components.footer import footer
+from buildcheck.components.notification import notifications_page # Import the UI component that displays notifications
+from buildcheck.state.user_state import UserState # Import global user state (which stores logged in user data like name, role, id, etc.)
+
+
+# 1. Define Notification State to track user info and hold fetched notifications
+class NotificationState(UserState):
```

Comment:
Instead of subclassing `UserState`, which may be disadvantageous, subclass rx.State


```suggestion
class NotificationState(rx.State):
```

See this comment for why https://github.com/ynx0/arch/pull/30#discussion_r2221322859